backend/src/main/java/com/github/stackscrubs/stuq/backend/service/SubjectService.java

```diff
@@ -0,0 +1,217 @@
+     * @return A list of teachers that teach the subject with the given subject code that takes place during the given term.
+     * @throws TermNotFoundException The term with the given ID is not registered in the database.
+     * @throws SubjectNotFoundException The subject code is not registered in the database.
+     */
+    public List<Teacher> getTeachers(TermId termId, String subjectCode) {
+        Term term = this.getTermOrThrow(termId);
+
+        return new ArrayList<>(
+            this.getSubjectOrThrow(new SubjectId(subjectCode, term)).getTeachers()
+        );
+    }
+
+    /**
+     * Gets the teaching assistants that assist the teachers in a certain subject.
+     * @param termId The ID of the term in which the subject takes place.
+     * @param subjectCode The subject's code.
+     * @return A list of teaching assistants that assist teachers in a certain subject
+     *         with the given subject code that takes place during the given term.
+     * @throws TermNotFoundException The term with the given ID is not registered in the database.
+     * @throws SubjectNotFoundException The subject code is not registered in the database.
+     */
+    public List<TeachingAssistant> getTeachingAssistants(TermId termId, String subjectCode) {
+        Term term = this.getTermOrThrow(termId);
+
+        return new ArrayList<>(
+            this.getSubjectOrThrow(new SubjectId(subjectCode, term)).getTeachingAssistants()
+        );
+    }
+
+    /**
+     * Gets assignments that are handed out to students in a certain subject.
+     * @param termId The ID of the term in which the subject takes place.
+     * @param subjectCode The subject's code.
+     * @return A list assignments that are handed out to students in a certain subject
+     *         with the given subject code that takes place during the given term.
+     * @throws TermNotFoundException The term with the given ID is not registered in the database.
+     * @throws SubjectNotFoundException The subject code is not registered in the database.
+     */
+    public List<Assignment> getAssignments(TermId termId, String subjectCode) {
+        Term term = this.getTermOrThrow(termId);
+
+        return new ArrayList<>(
+            this.getSubjectOrThrow(new SubjectId(subjectCode, term)).getAssignments()
+        );
+    }
+
+    /**
+     * Creates a new subject with a given subject code and name that takes place during the given term.
+     * @param termId The ID of the term in which the subject takes place.
+     * @param subjectCode The subject's code.
+     * @param subjectName The subject's full name.
+     * @throws SubjectAlreadyExistsException The subject already exists.
+     */
+    public void create(TermId termId, String subjectCode, String subjectName) {
+        Term term = this.termRepository.existsById(termId)
+                    ? this.termRepository.getById(termId)
+                    : this.termRepository.save(new Term(termId));
```

Comment:
we should just return not found if the term doesnt exist. it should be created on a different endpoint